### PR TITLE
Add azure storage account export target

### PR DESCRIPTION
**Pull request by cklingspor**
This PR prepares:

- `data_types`, `ignore_alloc_keys` & `rename_columns_config` to be read from ConfigMaps by reading them from file
- Business logic to use Service Principal and upload blob to Azure Storage Account container
- Factory pattern to target aws & azure blob storages. As of now, the code would not allow saving to local file system
- Testing extended accordingly
- Lots of thoughts what could be added to the code or where error handling could be improved
- Adjust README accordingly. But I wanted to checkin with you guys first.

### CODE REVIEW

**Comment by mattray** [comment]
I'll have to defer to @lmello and/or @sntxrr on this one

**Comment by lmello** [comment]
Thanks for your contribution. I am reviewing it.

**Comment by lmello** [comment]
@cklingspor 

I love the idea of moving data_types, ignore_alloc_keys & rename_columns_config to config maps. This is what I was planning to do to make this more parametrized.  

I like the idea of supporting both s3 and azure blob storage, but must still support local filesystem. 

Please let me know when you are ready for review, Since I am using this in my production environment I will need to extensively test it before merging.

File: src/opencost_parquet_exporter.py
Comment (lmello) [inline src/opencost_parquet_exporter.py:131]: Can you also parametrize includeIdle, idleByNode ?

File: src/opencost_parquet_exporter.py
Comment (lmello) [inline src/opencost_parquet_exporter.py:104]: Thanks for keeping backward compatibility.

File: src/opencost_parquet_exporter.py
Comment (lmello) [inline src/opencost_parquet_exporter.py:139]: This was not set before, it is better if we keep the previous behaviour and parametrize this in case you need to set it to true to your usecase.

File: src/opencost_parquet_exporter.py
Comment (lmello) [inline src/opencost_parquet_exporter.py:16]: add some test cases related to the data types, ignore_alloc_keys and rename_columns

**Review by lmello — CHANGES_REQUESTED** [review]
Thanks for the contribution, I really appreciate it. left some comments for you. 

let me know when you get it ready so I can test on my environment before merging.

**Comment by cklingspor** [comment]
@lmello : I've added  the env variables (was using that already in a private version. I've also updated the docs. I think you could run a test on your environment. Interested to hear your findings 😊

**Comment by lmello** [comment]
Sorry for the long delay. I am updating my opencost environment this week to version 1.110 and will be able to test this change next week. hope it is ok for you @cklingspor

**Comment by cklingspor** [comment]
No worries, works for me @lmello . Thanks!

**Comment by lmello** [comment]
@cklingspor sorry for the delay to test this. Trying to get some time at work. I hope to have a faster turn around on your next contributions.

File: src/opencost_parquet_exporter.py
Comment (lmello) [inline src/opencost_parquet_exporter.py:288]: Please update the Dockerfile file to support this.  
The dockerfile need major update for this PR also because of the new files under src/ used for azure. 

Remember to also change the examples kubernetes resource if the entry point changes.

**Review by lmello — CHANGES_REQUESTED** [review]

**Comment by lmello** [comment]
I liked that by adding the files we allow people to overwrite them on their docker images or using config maps.

**Comment by cklingspor** [comment]
Hi @lmello , 
was on a longer vacation. Thank you for the review. I'll take a look the upcoming week and submit changes.

File: src/opencost_parquet_exporter.py
Comment (cklingspor) [reply src/opencost_parquet_exporter.py:288]: Hi @lmello,
I've updated the PR. Was not 100% sure how you envisioned allowing for changes of `ENTRYPOINT` so please have look and let me know if you had something different in mind. I hope the changes address all requested changes.

**Comment by lmello** [comment]
Its good you added this, because I will need to deploy to gcp and you create the basis for supporting multiple providers, I am going to be adding the gcp support soon.

**Review by lmello — APPROVED** [review]

**Comment by lmello** [comment]
Merging this PR! Thanks for your contribution @cklingspor !!!!

**Pull request merged**
